ui/src/app/components/rename-modal.spec.tsx

```diff
@@ -60,3 +63,72 @@ describe('should show error if new name already exist', () => {
     }
   );
 });
+
+describe('RenameModal', () => {
+  let renameModal: RenameModal;
+
+  beforeEach(() => {
+    renameModal = new RenameModal({
+      existingNames: [],
+      oldName: '',
+      onCancel: () => {},
+      onRename: () => {},
+      resourceType: ResourceType.NOTEBOOK,
+    });
+  });
+
+  describe('validateNewName', () => {
+    it('should return no errors for valid new name', () => {
+      const errors = renameModal.validateNewName(
+        ['existingName.ipynb'],
+        'oldName.ipynb',
+        'newName.ipynb',
+        ResourceType.NOTEBOOK
+      );
+      expect(errors).toBeUndefined();
+    });
+
+    it('should return errors for duplicate new name', () => {
+      const errors = renameModal.validateNewName(
+        ['existingName.ipynb'],
+        'oldName.ipynb',
+        'existingName.ipynb',
+        ResourceType.NOTEBOOK
+      );
+      expect(errors).toBeDefined();
+      expect(errors.newName).toEqual(['New name already exists']);
+    });
+
+    it('should return errors for duplicate R file new name case insensitive', () => {
+      const errors = renameModal.validateNewName(
+        ['existingName.r'],
+        'oldName.r',
+        'existingName.R',
+        ResourceType.NOTEBOOK
+      );
+      expect(errors).toBeDefined();
+      expect(errors.newName).toEqual(['New name already exists']);
+    });
+
+    // Is this correct?
```

Comment:
arguable...

From a file system perspective (for some filesystems but not others!), it's fine to have these 2 distinct files.  But it's likely to lead to user confusion, so I think prohibiting this is a good choice.

Reply:
However I see at L121 that this has *no* error.  So something is inconsistent here.

Reply:
sorry that's typo, it should have been "should not return" I am allowing this in the current design. but like you said, I'm afraid this will overwrite the existing file, so I'll look into disabling it..

Reply:
actually the UI now disabled this, checking why the test doesn't reflect this.